app/controllers/api/openid_connect/discovery_controller.rb

```diff
@@ -0,0 +1,34 @@
+module Api
+  module OpenidConnect
+    class DiscoveryController < ApplicationController
+      def webfinger
+        jrd = {
+          links: [{
+            rel:  OpenIDConnect::Discovery::Provider::Issuer::REL_VALUE,
+            href: root_url
+          }]
+        }
+        jrd[:subject] = params[:resource] if params[:resource].present?
+        render json: jrd, content_type: "application/jrd+json"
+      end
+
+      def configuration
+        render json: OpenIDConnect::Discovery::Provider::Config::Response.new(
+          issuer:                                      root_url,
+          registration_endpoint:                       api_openid_connect_clients_url,
+          authorization_endpoint:                      new_api_openid_connect_authorization_url,
+          token_endpoint:                              api_openid_connect_access_tokens_url,
+          userinfo_endpoint:                           api_openid_connect_user_info_url,
+          jwks_uri:                                    File.join(root_url, "api", "openid_connect", "jwks.json"),
```

Comment:
Why is there no route helper here? You can set the format with `foo_url(format: "json")`